### PR TITLE
Add concept of parent and sub-topics, where sub-topics are added automatically when

### PR DESCRIPTION
**Pull request by Will-Howard**
Add the concept of parent and sub tags, where parent tags are applied automatically when a sub tag is applied. The initial use case is to make "AI Safety" a parent topic to "AI Alignment", "Artificial Intelligence"

The logic for when to apply a parent tag is:
 - when a subtag is applied to a post for the first time (i.e. there were no previous votes on it's releveance), apply the same vote to the parent tag
 - otherwise do nothing

### Link to parent and subtopics in tag pages

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1202780727457004) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/lib/collections/tags/custom_fields.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    viewableBy: ['guests'],
+    editableBy: ['sunshineRegiment', 'admins'],
+    insertableBy: ['sunshineRegiment', 'admins'],
+    label: "Parent Tag",
+    tooltip: "Parent tag which will also be applied whenever this tag is applied to a post for the first time",
+    group: formGroups.advancedOptions,
+    control: 'TagSelect',
+    onInsert: async (tag) => {
+      if (tag.parentTagId) {
+        // don't allow chained parent tag relationships
```
Comment (Will-Howard) [inline packages/lesswrong/lib/collections/tags/custom_fields.ts:24]: This is to avoid having to detect dependency cycles

**Comment by Xodarap** [comment]
I don't see the "+2 subtopics" UI bit. Did that change?

image

**Review by Xodarap — APPROVED** [review]
One minor comment, but otherwise this looks good to me.

**Comment by Will-Howard** [comment]
> EDIT: yes, you said that changed. But I kind of liked it – is there a reason you deleted it?

Ruby said he thought it was unnecessary and I didn't have a strong preference so I removed it. I can add it back in in another PR if you are strongly pro. If I added it back in it would now say "+ Parent Topic" because in the first version I was adding subtopics when a parent topic was added, now it's the other way round

**Pull request merged**